src/main/java/org/jenkinsci/plugins/codesonar/services/AlertsService.java

```
@@ -0,0 +1,110 @@
+                    continue;
+                } else if(e.getStatus() == 500) {
+                    /*
+                     * A 500 response is returned anytime the request specifies an alert kind that is out
+                     * of range with respect to all currently supported alert kinds.
+                     * This makes status 500 the right one to establish when to stop looping through kinds.
+                     */
+                    LOGGER.log(Level.INFO, MessageFormat.format("Stop looping through alert kinds at kind id {0}", kind), e);
+                    break;
+                } else {
+                    /*
+                     * A server-side error prevented the response from being satisfied, try skipping current alert kind
+                     */
+                    LOGGER.log(Level.INFO, "Found an unexpected error in the response, skipping current alert kind", e);
+                    continue;
+                }
+            } catch(CodeSonarPluginException e) {
+                /*
+                 * An generic error prevented the response from being received, try skipping current alert kind
+                 */
+                LOGGER.log(Level.INFO, "Unexpected error querying the hub, skipping current alert kind", e);
+                continue;
+            }
+    
+            Gson gsonDeserializer = new Gson();
+            CodeSonarAlertData alertData = null;
+            try {
+                alertData = gsonDeserializer.fromJson(new InputStreamReader(jsonContent, StandardCharsets.UTF_8), CodeSonarAlertData.class);
+                LOGGER.log(Level.INFO, "response alertData={0}", alertData.toString());
+            } catch(JsonSyntaxException e) {
+                LOGGER.log(Level.WARNING, "failed to parse JSON response. %nException: {0}%nStack Trace: {1}", new Object[] {e.getMessage(), Throwables.getStackTraceAsString(e)});
+                return null;
+            }
```

I don't think we should return null.  I think we should throw an exception.